src/shared/Microsoft.Git.CredentialManager/GitConfiguration.cs

```diff
@@ -3,117 +3,121 @@
         /// </summary>
+        /// <param name="level">Filter to the specific configuration level.</param>
         /// <param name="nameRegex">Configuration entry name regular expression.</param>
         /// <param name="valueRegex">Regular expression to filter which variables we're interested in. Use null to indicate all.</param>
         /// <returns>All values of the multivar configuration entry.</returns>
-        IEnumerable<string> GetRegex(string nameRegex, string valueRegex);
+        IEnumerable<string> GetRegex(GitConfigurationLevel level, string nameRegex, string valueRegex);
 
         /// <summary>
         /// Set a multivar configuration entry value.
         /// </summary>
+        /// <param name="level">Filter to the specific configuration level.</param>
         /// <param name="nameRegex">Configuration entry name regular expression.</param>
         /// <param name="valueRegex">Regular expression to indicate which values to replace.</param>
         /// <param name="value">Configuration entry value.</param>
         /// <remarks>If the regular expression does not match any existing entry, a new entry is created.</remarks>
-        void ReplaceAll(string nameRegex, string valueRegex, string value);
+        void ReplaceAll(GitConfigurationLevel level, string nameRegex, string valueRegex, string value);
 
         /// <summary>
         /// Deletes one or several entries from a multivar.
         /// </summary>
+        /// <param name="level">Filter to the specific configuration level.</param>
         /// <param name="name">Configuration entry name.</param>
         /// <param name="valueRegex">Regular expression to indicate which values to delete.</param>
-        void UnsetAll(string name, string valueRegex);
+        void UnsetAll(GitConfigurationLevel level, string name, string valueRegex);
     }
 
     public class GitProcessConfiguration : IGitConfiguration
     {
         private readonly ITrace _trace;
         private readonly GitProcess _git;
-        private readonly GitConfigurationLevel? _filterLevel;
 
-        internal GitProcessConfiguration(ITrace trace, GitProcess git, GitConfigurationLevel filterLevel = GitConfigurationLevel.All)
+        internal GitProcessConfiguration(ITrace trace, GitProcess git)
         {
             EnsureArgument.NotNull(trace, nameof(trace));
             EnsureArgument.NotNull(git, nameof(git));
 
             _trace = trace;
             _git = git;
-            _filterLevel = filterLevel;
         }
 
-        public void Enumerate(GitConfigurationEnumerationCallback cb)
+        public void Enumerate(GitConfigurationLevel level, GitConfigurationEnumerationCallback cb)
         {
-            string level = GetLevelFilterArg();
-            using (Process git = _git.CreateProcess($"config --null {level} --list"))
+            string levelArg = GetLevelFilterArg(level);
+            using (Process git = _git.CreateProcess($"config --null {levelArg} --list --show-scope"))
```

Comment:
I agree that we require the information about the scope. The extensive changes required by that make be a _bit_ uneasy, but I think that cannot be helped.

Reply:
Yeah, this is very much a big change to how we parse config output. But it's the only way I can see to do this whilst also adding the required scope data.